atlasdb-cassandra/src/main/java/com/palantir/atlasdb/cassandra/CassandraKeyValueServiceConfig.java

```diff
@@ -59,7 +60,22 @@ public int unresponsiveHostBackoffTimeSeconds() {
 
     public abstract Optional<CassandraCredentialsConfig> credentials();
 
-    public abstract boolean ssl();
+    /**
+     * A boolean declaring whether or not to use ssl to communicate with cassandra.
+     * This configuration value is deprecated in favor of using sslConfiguration.  If
+     * true, read in trust and key store information from system properties unless
+     * the sslConfiguration object is specified.
+     */
+    @Deprecated
+    public abstract Optional<Boolean> ssl();
+
+    /**
+     * An object for specifying ssl configuration details.  The lack of existence of this
+     * object implies ssl is not to be used to connect to cassandra.
+     *
+     * The existence of this object overrides any configuration made via the ssl config value.
+     */
+    public abstract Optional<SslConfiguration> sslConfiguration();
```

Comment:
@ash211 I feel like if that's the way we intend to think about it broadly (as Mark describes) then I'd like to fit that model.  Happy to talk offline if you think that model is wrong.